### DISCUSSION TITLE
Automatic unbinding after idle?

### DISCUSSION THREAD

**Discussion by davidstackio** [vuejs/vuefire#1428] [Questions and Help]
When using `useCollection` with the `target` option inside a Pinia store using the Composition API, the data initially is shown in the UI. However, after a couple minutes of being idle, there is nothing showing any more. The store seems to automatically unbind. Only way to fix it is by refreshing the page.

Is this expected behavior?

### store.ts
```ts
...

const activityColRef = computed(() => {
  if (currentUser.value && defaultLocation.value && defaultPeriod.value) {
    return collection(
      db,
      "users",
      currentUser.value.uid,
      "locations",
      defaultLocation.value.id,
      "periods",
      defaultPeriod.value.id,
      "activities"
    );
  }
});
const activityQuery = computed(() => {
  if (activityColRef.value) {
    return query(activityColRef.value, orderBy("lastUpdated", "desc"));
  }
});
const activityCol = ref<ActivityDoc[]>([]);
useCollection<ActivityDoc>(activityQuery.value, {
  target: activityCol,
});

...
```

**Comment by posva** [comment]
I'm not aware of such behavior. Have you been able to confirm from Firebase docs that this is intended? In VueFire, the subscription is removed when the component unmounts.

Unrelated but you can write

```ts
const activityCol = useCollection<ActivityDoc>(activityQuery);
```

**Reply by davidstackio** [reply]
@posva I have not, which is why I asked here.

The reason I wrote it that way was because I had some code later that fills the store on initial load and without using `target` it seemed to break the real time syncing.

In the end I couldn't figure it out and just needed it working so I opted to remove VueFire from my Pinia store and am using the Firebase native `onSnapshot` instead. See https://github.com/vuejs/vuefire/discussions/1426#discussioncomment-6969646
